library/ssl_tls13_client.c

```diff
@@ -3875,6 +3875,9 @@ static int ssl_hrr_postprocess( mbedtls_ssl_context* ssl,
 
     ssl->handshake->hello_retry_requests_received++;
 
+    MBEDTLS_SSL_DEBUG_MSG( 4, ( "Reset transform_out" ) );
+    ssl->transform_out = NULL;
```

Comment:
Uh, be careful, this isn't Python :-) We have to make sure we're not leaving memory behind, which I fear is the case here.